### PR TITLE
Exit with an error if a nonexistent command is attempted

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

File: build_runner/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 - Improved the layout of `/$perf`, especially after browser window resize.
 
+- `pub run build_runner` exits with a error when invoked with an unsupported
```
Comment (natebosch) [inline build_runner/CHANGELOG.md]: Remove the extra blank line above

File: build_runner/bin/build_runner.dart [hunk 2]
```diff
@@ -23,6 +25,16 @@ Future<Null> main(List<String> args) async {
     ..addCommand(new _GenerateBuildScript());
   var parsedArgs = commandRunner.parse(args);
   var commandName = parsedArgs.command?.name;
+
+  if (parsedArgs.rest.isNotEmpty) {
```
Comment (natebosch) [inline build_runner/bin/build_runner.dart]: we detect this today with `commandName == null` - should we continue to do that and remove that condition below?

Comment (kevmoo) [reply build_runner/bin/build_runner.dart]: `null` is no command. I think we still want to print help and exit code 0 there.

File: build_runner/CHANGELOG.md (hunk 1 above)
Comment (kevmoo) [reply build_runner/CHANGELOG.md]: ack

**Pull request merged**